### ISSUE TITLE
Epub support

### ISSUE THREAD

**Issue by whalehub** [gotson/komga#69]
Is this a planned feature for Komga?

**Comment by gotson** [comment]
Hi, if you can provide some sample epub files then I can have a look.

Basic support should be easy, as epubs are zip files. 

But they also have html with potential embedded images.

**Comment by whalehub** [comment]
@gotson You can download some .epub files here: https://filerun.datahoarder.dev/wl/?id=ZLAAgYOyr6L3qeZ0z3ViNLgFbXWlB1W1

**Comment by gotson** [comment]
I was thinking of `epub` with images in it. I have no plan to handle eBooks !

**Comment by gerroon** [comment]
So epubs are never going to be supported? I am not a dev so I cant be any use in this regard but just curious about this.

**Comment by gotson** [comment]
Komga will not support **ebooks** in any foreseeable future, no.

`epub` is a file format, mostly used for ebooks. I know some comics are packaged in the `epub` format, but I would need sample comics in `epub` to add support for it (if that is the case please open a new issue for that).

**Comment by The7thSage** [comment]
I think I may have an idea about this, so I'll chime in, even at the risk of looking like a layman. 
Just like .cbz and .cbr files, .epub files behave like a glorified zip file as well. More specifically it is one or more .html/.xhtml files, their resources, metadata files, and info for dumb e-readers to find these files. 

What this means, is two things,
1.  Images can be named anything, as long as the html-code places them in the right order
2.  Image files are in no proprietary format, so they can be copied/read at will, even extracted, edited and rezipped most things aren't needed.

So there can be two solutions, the lossy/shortcut way or programing in some way to read the html file and order. The latter seems to be too much work. The only imaginably 'easy' way is to add zip as an extension to be read like a zip or cbz The problem is it may appear "buggy" when the program reading files in an incorrect page order, but correct named-order.  This is something that would be monumental to fix, as hypothetically any one publisher can do whatever he wants, as long as it works.

Long story short, Komga very likely already has support for (most) your epub-comics. Push-comes-to shove, you might just have to shell out some HDD space to make a copy renamed from "comic.epub" to "comic.zip". If that doesn't work (or produces out-of-order pages, make a cbz (also just a renamed zip). It might not be easy, but it works.

**Comment by The7thSage** [comment]
As gotson posted earlier, 

> epubs are zip files
So I'll share what I do to read my epubs on komga, this whole time.
So just copy and rename your comic-book epubs into zip (comic.epub becomes comic.zip) and try them out.
If the order of the pages is wrong, (and you are on windows) and you don't want to go through trouble organizing them by hand, install Calibre, click edit book, select all the images, and bulk rename (clicking on "sort by book order) and wallah, it has been done. You can even "batch convert" them to zip, if you cant find a way to do that.

If you have a text-based Ebook, also install Calibre or another program that can batch convert your epubs to pdf. It's not rocket science.  (for the calibre users I use a page size of my monitor in order to get a sufficiently large 19:6 portrait image.)

**Comment by gotson** [comment]
I have created a new issue for comics in epub (#119). I lock this issue since i have no plan as of now to support ebooks in Komga.
